### PR TITLE
Implement service layer

### PR DESCRIPTION
**Pull request by cosletmihai**

### CODE REVIEW

**Comment by cosletmihai** [comment]
There is some code duplication from the `rdf-differ` in the `FusekiSPARQLAdapter`

File: tests/unit/test_adapters_sparql.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    sparql_wrapper_adapter = helper_sparql_wrapper(http_client=fake_requests)
+
+    with pytest.raises(FusekiException) as e:
+        _ = sparql_wrapper_adapter.get_dataset('dataset')
+
+    assert exception_message in str(e)
+
+
+def test_sparql_wrapper_adapter_upload_file_success(tmpdir, fake_requests):
```
Comment (costezki) [inline tests/unit/test_adapters_sparql.py:38]: these tests are from the other project, I suppose.

File: tests/unit/test_services_handlers.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+from fingerprinter.services.handlers import fingerprint_file, upload_file_to_dataset
+
+
+def test_upload_file_to_dataset(tmpdir, fake_sparql_adapter):
+    file = tmpdir.join('file.ttl')
+
+    with upload_file_to_dataset('dataset', str(file), fake_sparql_adapter):
+        pass
+
+    assert fake_sparql_adapter.actions[0] == ('CREATE', 'dataset')
```
Comment (costezki) [inline tests/unit/test_services_handlers.py:19]: please show me how you implement the new "Domain" logic

**Review by costezki — APPROVED** [review]
well done

File: docker/.env [hunk 3]
```diff
@@ -1,4 +1,11 @@
 FINGERPRINTER_API_PORT=4020
 FINGERPRINTER_UI_PORT=8020
```
[the remark this answers is not in the document]
Comment (cosletmihai) [reply docker/.env:2]: I agree about these ones, but the fuseki ones are, presumably, all services are going to use the same fuseki server, so doesn't it make more sense to not "brand" them with the service that uses the envs? (`RDF_FINGERPRINTER_FUSEKI_PORT`)

File: fingerprinter/entrypoints/api/handlers.py [hunk 4]
```diff
@@ -15,20 +15,24 @@
 logger = logging.getLogger(__name__)
 
 
-def fingerprint_file(data_file: FileStorage) -> tuple:
+def fingerprint_file(body: dict, data_file: FileStorage, report_extension: str = 'html') -> tuple:
```
[the remark this answers is not in the document]
Comment (cosletmihai) [reply fingerprinter/entrypoints/api/handlers.py:18]: this functionality is not yet implemented in this issue, will be done in the next PR

File: fingerprinter/entrypoints/api/openapi/fingerprinter.yml [hunk 5]
```diff
@@ -29,26 +29,39 @@ components:
...
-          description: The endpoint to validate
-      required:
-        - sparql_endpoint_url
+      allOf:
+        - $ref: '#components/schemas/BaseFingerprinter'
+        - type: object
+          properties:
+            sparql_endpoint_url:
+              type: string
+              description: The endpoint to validate
```
[the remark this answers is not in the document]
Comment (cosletmihai) [reply fingerprinter/entrypoints/api/openapi/fingerprinter.yml:62]: I don't understand what you mean, could you please elaborate?

File: fingerprinter/services/handlers.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+import uuid
+from contextlib import contextmanager
+
+from fingerprint.service_layer.handlers import generate_endpoint_fingerprint_report
+
+from fingerprinter import config
+from fingerprinter.adapters.sparql_adapter import AbstractSPARQLAdapter
+
+
+@contextmanager
```
[the remark this answers is not in the document]
Comment (cosletmihai) [reply fingerprinter/services/handlers.py:20]: this functionality doesn't seem to fall under the UoW realm, it just uses the context manager as an easier way to solve the problem of uploading and cleaning up the fuseki triplestore

File: fingerprinter/services/handlers.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    """
+        A context manager that creates a dataset, uploads a file into it, and finally deletes the dataset
+    :param dataset: dataset name
+    :param file_path: location of the file to be uploaded
+    :param sparql_adapter: adapter used to perform triplestore operations
+    """
+    try:
+        sparql_adapter.create_dataset(dataset)
+        sparql_adapter.upload_file(dataset, file_path)
+        yield
```
[the remark this answers is not in the document]
Comment (cosletmihai) [reply fingerprinter/services/handlers.py:31]: yes, there is nothing that needs to be `yield`ed in with this context manager

File: tests/steps/test_ui_fingerprinting.py [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+@scenario('../features/ui_fingerprinting.feature', 'Fingerprint user provided data')
+def test_fingerprint_user_provided_data():
+    """Fingerprint user provided data."""
+
+
+@given('a <fingeprinting_type>')
+def a_fingeprinting_type(fingeprinting_type):
+    """a <fingeprinting_type>."""
+    raise NotImplementedError
```
[the remark this answers is not in the document]
Comment (cosletmihai) [reply tests/steps/test_ui_fingerprinting.py:20]: the ui functionality is not yet implemented

File: tests/unit/test_adapters_sparql.py (hunk 1 above)
Comment (cosletmihai) [reply tests/unit/test_adapters_sparql.py:38]: no, these tests are for the current project

File: tests/unit/test_services_handlers.py (hunk 2 above)
Comment (cosletmihai) [reply tests/unit/test_services_handlers.py:19]: we don't have domain logic in this project

File: fingerprinter/entrypoints/api/handlers.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (cosletmihai) [reply fingerprinter/entrypoints/api/handlers.py:18]: I've added a [jira ticket](https://meaningfy.atlassian.net/browse/MDR-265?atlOrigin=eyJpIjoiYjJmODQ0NGYwMGJiNGY4YWI2Njc5M2M3ZDY1NTQxNGQiLCJwIjoiaiJ9) to update it on the `rdf-validator-ws` as well

**Pull request merged**